pkg/config/config.go

```diff
@@ -38,6 +40,13 @@ type NSXOperatorConfig struct {
 	*HAConfig
 }
 
+func (operatorConfig *NSXOperatorConfig) HAEnabled() bool {
+	if operatorConfig.EnableHA == nil || *operatorConfig.EnableHA == true {
```

Comment:
How about set default value in NewNSXOpertorConfig?
It semes not good idea after NewNSXOpertorConfig, it still have some filed not initialized.
So you could also using EnableHA directly instead of HAEnabled().
and EnableHA type could also be bool.